src/docs/guide/usage/formatter/migrate-from-prettier.md

```diff
@@ -0,0 +1,228 @@
+  "printWidth": 80,
+}
+```
+
+If you have a basic `.prettierrc` file, you can simply rename the file with `mv .prettierrc .oxfmtrc.jsonc`.
+
+If you are using YAML or JavaScript to configure Prettier, you will need to convert the configuration to JSON format manually, although for simple configs this should be trivial.
+
+### `prettierrc.js`
+
+Here's an example of migrating a `prettierrc.js` file.
+
+Before:
+
+```js
+module.exports = {
+  singleQuote: true,
+  jsxSingleQuote: true,
+};
+```
+
+After (`.oxfmtrc.jsonc`):
+
+```jsonc
+{
+  "$schema": "./node_modules/oxfmt/configuration_schema.json",
+  "singleQuote": true,
+  "jsxSingleQuote": true,
+  "printWidth": 80,
+}
+```
+
+### `prettierrc.yaml`
+
+Here's an example of migrating a `prettierrc.yaml` file.
+
+Before:
+
+```yaml
+trailingComma: "es5"
+tabWidth: 4
+semi: false
+singleQuote: true
+```
+
+After (`.oxfmtrc.jsonc`):
+
+```jsonc
+{
+  "$schema": "./node_modules/oxfmt/configuration_schema.json",
+  "trailingComma": "es5",
+  "tabWidth": 4,
+  "semi": false,
+  "singleQuote": true,
+  "printWidth": 80,
+}
+```
+
+## Step 4: Update Formatting Scripts
+
+Update any formatting scripts you currently have, for example in `package.json`, shell scripts, or pre-commit scripts.
+
+### `package.json` scripts
+
+```diff
+{
+  "scripts": {
+-   "format": "prettier --write .",
++   "format": "oxfmt",
+-   "format:check": "prettier --check ."
++   "format:check": "oxfmt --check"
+  }
+}
+```
+
+### CI Workflows
+
+Update any CI workflows that run Prettier, particularly `prettier --check`.
+
+```diff
+  - name: Check formatting
+-   run: yarn prettier --check .
++   run: yarn oxfmt --check
+```
+
+### Git Hooks (e.g. husky, lint-staged)
+
+```diff
+"lint-staged": {
+- "*": "prettier --write --no-error-on-unmatched-pattern"
++ "*": "oxfmt --no-error-on-unmatched-pattern"
+}
+```
+
+## Step 5: Run formatter
+
+Run Oxfmt on your codebase to check for any changes and ensure that the configuration was migrated correctly:
+
+```sh
+# Your script specified in Step 4
+npm run format
+```
+
+If you no longer need Prettier, you can uninstall for now.
+
+## Done!
+
+You have now migrated to Oxfmt :)
+
+Please see the section below for any additional, optional steps you may need to take.
+
+These are only applicable for some setups, so skip them if they don't apply to you.
+
+### Update editor integrations
+
+**TODO: Clarify the status of VS Code integration, and also add a note for potentially using Oxfmt in JetBrains IDEs + others via the LSP server.**
+
+**Note that the oxc VS Code extension includes Oxfmt support, but it is experimental and incomplete**. It may not fully work yet, so please be mindful of this before migrating.
+
+If you have any editor integrations for Prettier, update them to use Oxfmt instead. For example, update `.vscode/settings.json` to use Oxfmt:
+
+```diff
+{
+- "editor.defaultFormatter": "esbenp.prettier-vscode"
++ "editor.defaultFormatter": "oxc.oxc-vscode"
```

Comment:
Since oxfmt doesn't yet support non-JS files, it's not practical to replace the prettier extension entirely with it. I think it's better to suggest enabling it for specific languages (for now, `[javascript]`, `[javascriptreact]`, `[typescript]`, and `[typescriptreact]`).

And we still need `"oxc.fmt.experimental": true,` at the moment, if I'm not mistaken.

Reply:
I've removed this section in favor of a link to the FAQ question about editor integration, in the future we'll probably have a dedicated editor integration page :)